src/app.js

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2015 Colin Clark
+
+Licensed under the 3-Clause "New" BSD license.
+You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the 3-Clause "New" BSD License at
+https://github.com/colinbdclark/infusion-electron/raw/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    electron = fluid.registerNamespace("electron");
+
+// Monkey-patch fluid.log for Electron's broken console.log.
+fluid.doLog = function (args) {
+    if (typeof (console) !== "undefined") {
+        if (console.debug) {
+            console.debug(args.join(" "));
+        } else if (typeof (console.log) === "function") {
+            console.log(args.join(" "));
+        }
+    }
+};
+
+
+fluid.defaults("electron.app", {
+    gradeNames: "fluid.modelComponent",
+
+    members: {
+        app: require("app")
+    },
+
+    commandLineSwitches: {},
+
+    windowListeners: {
+        "onClose": "electron.app.quitOnAllClosed()"
+    },
+
+    events: {
+        onReady: null,
+        onAllWindowsClosed: null
+    },
+
+    listeners: {
+        onCreate: [
+            {
+                funcName: "electron.app.setCommandLineSwitches",
+                args: ["{that}.app", "{that}.options.commandLineSwitches"]
+            },
+            {
+                "this": "{that}.app",
+                method: "on",
+                args: ["window-all-closed", "{that}.events.onAllWindowsClosed.fire"]
+            },
+            {
+                "this": "{that}.app",
+                method: "on",
+                args: ["ready", "{that}.events.onReady.fire"]
+            }
+        ],
+
+        onAllWindowsClosed: [
+            "electron.app.handleAllWindowsClosed({that}.app)"
+        ]
+    }
+});
+
+electron.app.handleAllWindowsClosed = function (app) {
+    if (process.platform !== "darwin") {
+        app.quit();
+    }
+};
+
+electron.app.setCommandLineSwitches = function (app, commandLineSwitches) {
+    fluid.each(commandLineSwitches, function (value, switchName) {
+        if (value === null) {
```

Comment:
This might be an area of improvement for our API. Command line switches in Electron may have arguments, but they might not:

https://github.com/atom/electron/blob/master/docs/api/chrome-command-line-switches.md

When I implemented this API in our wrapper in a more data-oriented way, I decided to use a hash of switch/value pairs, with <code>null</code> indicating no value.

Here's an example of how it's used in practice:

https://github.com/colinbdclark/lou/blob/master/js/app/lou.js#L13-L16

Do you have any thoughts on a better strategy for this?